interfaces/builtin/gpio_memory_control.go

```diff
@@ -0,0 +1,51 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+const gpioMemoryControlSummary = `allows write access to all gpio memory`
+
+const gpioMemoryControlBaseDeclarationSlots = `
+  gpio-memory-control:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
+`
+
+const gpioMemoryControlConnectedPlugAppArmor = `
+# Description: Allow writing to /dev/gpiomem on kernels that provide it.
+#
+/dev/gpiomem rw,
```

Comment:
Don't you need `m` as well for mmap?

Reply:
Honestly I don't know. This worked for my purposes, which given my rudimentary understanding of the lib I'm using uses mmap, but it's possible they're falling back (note that I see no denials, however).

Note also that `physical-memory-control` does not use `m` for `/dev/mem`: https://github.com/snapcore/snapd/blob/master/interfaces/builtin/physical_memory_control.go#L41 .

Reply:
'm' is only needed when using mmap() with PROT_EXEC.